spring-kafka/src/main/java/org/springframework/kafka/core/reactive/ReactiveKafkaConsumerTemplate.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.core.reactive;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.consumer.OffsetAndTimestamp;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.kafka.receiver.KafkaReceiver;
+import reactor.kafka.receiver.ReceiverOptions;
+import reactor.kafka.receiver.ReceiverRecord;
+import reactor.kafka.sender.TransactionManager;
+
+/**
+ * Reactive kafka consumer operations implementation.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Mark Norkin
+ */
+public class ReactiveKafkaConsumerTemplate<K, V> implements ReactiveKafkaConsumerOperations<K, V> {
+	private final KafkaReceiver<K, V> kafkaReceiver;
+
+	public ReactiveKafkaConsumerTemplate(ReceiverOptions<K, V> receiverOptions) {
+		this.kafkaReceiver = KafkaReceiver.create(receiverOptions);
+	}
+
+	@Override
+	public Flux<ReceiverRecord<K, V>> receive() {
+		return this.kafkaReceiver.receive();
+	}
+
+	@Override
+	public Flux<Flux<ConsumerRecord<K, V>>> receiveAutoAck() {
+		return this.kafkaReceiver.receiveAutoAck();
+	}
+
+	@Override
+	public Flux<ConsumerRecord<K, V>> receiveAtmostOnce() {
+		return this.kafkaReceiver.receiveAtmostOnce();
+	}
+
+	@Override
+	public Flux<Flux<ConsumerRecord<K, V>>> receiveExactlyOnce(TransactionManager transactionManager) {
+		return this.kafkaReceiver.receiveExactlyOnce(transactionManager);
+	}
+
+	@Override
+	public <T> Mono<T> doOnConsumer(Function<Consumer<K, V>, ? extends T> function) {
+		return this.kafkaReceiver.doOnConsumer(function);
+	}
+
+	public Mono<Set<TopicPartition>> assignment() {
```

Comment:
This is somehow wrong from the reactive API perspective: if it is a `Collection` underneath let's consider to use `Flux.fromIterable()` instead of ` Mono<Set<?>>`!

Reply:
Done. 
Also did similar change to `ReactiveKafkaProducerTemplate::partitionsFromProducerFor` and for several api's on consumer template side.

Also there are several api's left with Mono over Map, for example there is `ReactiveKafkaProducerTemplate::metricsFromProducer` with return type `Mono<? extends Map<MetricName, ? extends Metric>>`, should we consider changing it to `Flux<? extends Map.Entry<MetricName, ? extends Metric>>` and perform these change for similar api's, WDYT ?